includes/payment-methods/class-wc-gateway-stripe-sepa.php

```diff
@@ -117,6 +118,8 @@ public function __construct() {
 
 		add_action( 'woocommerce_update_options_payment_gateways_' . $this->id, array( $this, 'process_admin_options' ) );
 		add_action( 'wp_enqueue_scripts', array( $this, 'payment_scripts' ) );
+        
+		add_action( 'woocommerce_order_status_changed', array( $this, 'maybe_activate_subscriptions_early'), 10, 4 );
```

Comment:
Original functionality:

**Complete Order + Active subscription -> Order set to `on-hold`**

This results in the Subscription being set to `on-hold`

<hr />

After applying the PR:

**Complete Order + Active subscription -> Order set to `on-hold`**

This results in the Subscription still being active. In fact, the subscription item is always set to active even if the merchant previously modified the Subscription status to be something else.

Reply:
I *think* if we want to handle this a bit more gracefully we probably need to add an action that's called after a SEPA subscription payment is completed. Well, it doesn't need to be an action actually. It sort of depends on how extensible we want to make this functionality.

Once we have this in place we can make sure we only update the subscription status when the following conditions are fulfilled:

1. The payment is for a subscription
1. The payment has been captured
2. The payment is pending
3. SEPA was used for the payment

That should make the detection here a bit more intelligent, and it won't go off every time an order's status is changed to `on-hold`.

Reply:
I think you're on to something I hadn't considered here. Yes, this should only run if the order was paid for via SEPA. Checking the ~ time that the order was placed/created would be useful too. It would need to run for parent and renewal orders, not just through the standard `woocommerce_thankyou` hook.

Reply:
I hadn't considered the renewal orders 🤔 

I think the best solution to this is still to append this functionality to the payment processing in the SEPA subs compat class instead of relying on hooks.

I'll work on that and see where that ends up 🙂